examples/factories/various_factories_usage.rb

```diff
@@ -0,0 +1,171 @@
+    cb.name = client.wrapper.string(
+      "Interplanetary Budget #{(Time.new.to_f * 1000).to_i}",
+    )
+
+    cb.delivery_method = :STANDARD
+    cb.amount_micros = client.wrapper.int64(500000)
+  end
+
+  return_budget = client.service.campaign_budget.mutate_campaign_budgets(
+    customer_id,
+    [campaign_budget_operation]
+  )
+
+  campaign_budget_resource_name = return_budget.results.first.resource_name
+
+  # Create campaign.
+  campaign = client.resource.campaign
+  campaign.name = client.wrapper.string(
+    "Interplanetary Cruise #{(Time.new.to_f * 1000).to_i}",
+  )
+  campaign.advertising_channel_type = :SEARCH
+
+  # Recommendation: Set the campaign to PAUSED when creating it to prevent
+  # the ads from immediately serving. Set to ENABLED once you've added
+  # targeting and the ads are ready to serve.
+  campaign.status = :PAUSED
+
+  campaign.manual_cpc = client.resource.manual_cpc
+  campaign.campaign_budget = client.wrapper.string(
+    campaign_budget_resource_name,
+  )
+
+  campaign.network_settings = client.resource.network_settings do |ns|
+    ns.target_google_search = client.wrapper.bool(true)
+    ns.target_search_network = client.wrapper.bool(true)
+    ns.target_content_network = client.wrapper.bool(false)
+    ns.target_partner_search_network = client.wrapper.bool(false)
+  end
+
+  # Alternate non-block style, where we pass the campaign object we built
+  # up
+  campaign_operation = client.operation.create_resource.campaign(campaign)
+  campaign_service = client.service.campaign
+
+  # Add the campaign.
+  response = campaign_service.mutate_campaigns(
+    customer_id,
+    [campaign_operation],
+  )
+  campaign_resource_name = response.results.first.resource_name
+  puts "Created campaign #{campaign_resource_name}"
+
+  # when we fetch campaigns from the API, they have resource names populated,
+  # but because we just created this one, it isn't populated, so this will
+  # let us work with the campaign we just created as an updatable campaign
+  campaign.resource_name = campaign_resource_name
+
+  update_operation = client.operation.update_resource.campaign(campaign) do
+    campaign.name = client.wrapper.string(
+      "A different interplanetary Cruise #{(Time.new.to_f * 1000).to_i}",
+    )
+  end
+
+  campaign_service.mutate_campaigns(customer_id, [update_operation])
+
+  # Finally remove the campaign
+  remove_op = client.operation.remove_resource.campaign(campaign_resource_name)
+  campaign_service.mutate_campaigns(customer_id, [remove_op])
+
+  # updates also work with only a resource name, so let's pull one out and
+  # then update it
+  ga_service = client.service.google_ads
+  res = ga_service.search(customer_id, <<~EOQUERY)
+  select campaign.resource_name, campaign.name from campaign limit 1
+  EOQUERY
+
+  campaign_resource_name = res.first.campaign.resource_name
+
+  # note here, we have to pass the campaign (`camp`) as a block arg, because
```

Comment:
Mention that the field mask is created and applied automatically based on what's set in the block.